libs/core/src/lib/skeleton/components/skeleton.component.ts

```diff
@@ -0,0 +1,71 @@
+import { ChangeDetectionStrategy, Component, HostBinding, Input, ViewEncapsulation } from '@angular/core';
+
+export type SkeletonType = 'circle' | 'rectangle' | 'text';
+export type SkeletonWidth = 'rand' | string;
+export type SkeletonHeight = 'auto' | string;
+
+@Component({
+    selector: 'fd-skeleton',
+    template: ``,
```

Comment:
maybe you could also add content projection here while making the content fully transparent? that will allow to have shapes based on fixed text size (e.g. exactly the width of `lorem ipsum` following the text size in the particular container)


Something like
```
<fd-skeleton width="inherit">lorem ipsum</fd-skeleton>


fd-skeleton.inherited-width {
    width: fit-content;
    color: transparent;
}
```